tests/by-util/test_df.rs

```diff
@@ -423,6 +423,79 @@ fn test_block_size_with_suffix() {
     assert_eq!(get_header("1GB"), "1GB-blocks");
 }
 
+#[test]
+fn test_block_size_from_env() {
+    fn get_header(env_var: &str, env_value: &str) -> String {
+        let output = new_ucmd!()
+            .arg("--output=size")
+            .env(env_var, env_value)
+            .succeeds()
+            .stdout_move_str();
+        output.lines().next().unwrap().to_string()
+    }
+
+    assert_eq!(get_header("DF_BLOCK_SIZE", "111"), "111B-blocks");
+    assert_eq!(get_header("BLOCK_SIZE", "222"), "222B-blocks");
+    assert_eq!(get_header("BLOCKSIZE", "333"), "333B-blocks");
+}
+
+#[test]
+fn test_block_size_from_env_precedences() {
+    fn get_header(one: (&str, &str), two: (&str, &str)) -> String {
+        let (k1, v1) = one;
+        let (k2, v2) = two;
+        let output = new_ucmd!()
+            .arg("--output=size")
+            .env(k1, v1)
+            .env(k2, v2)
+            .succeeds()
+            .stdout_move_str();
+        output.lines().next().unwrap().to_string()
+    }
+
+    let df_block_size = ("DF_BLOCK_SIZE", "111");
+    let block_size = ("BLOCK_SIZE", "222");
+    let blocksize = ("BLOCKSIZE", "333");
+
+    assert_eq!(get_header(df_block_size, block_size), "111B-blocks");
+    assert_eq!(get_header(df_block_size, blocksize), "111B-blocks");
+    assert_eq!(get_header(block_size, blocksize), "222B-blocks");
+}
+
+#[test]
+fn test_precedence_of_block_size_arg_over_env() {
+    let output = new_ucmd!()
+        .args(&["-B", "999", "--output=size"])
+        .env("DF_BLOCK_SIZE", "111")
+        .succeeds()
+        .stdout_move_str();
+    let header = output.lines().next().unwrap().to_string();
+
+    assert_eq!(header, "999B-blocks");
+}
+
+#[test]
+fn test_invalid_block_size_from_env() {
+    let output = new_ucmd!()
+        .arg("--output=size")
+        .env("DF_BLOCK_SIZE", "invalid")
+        .succeeds()
+        .stdout_move_str();
+    let header = output.lines().next().unwrap().to_string();
+
+    assert_eq!(header, "1K-blocks");
+
+    let output = new_ucmd!()
+        .arg("--output=size")
+        .env("DF_BLOCK_SIZE", "invalid")
+        .env("BLOCK_SIZE", "222")
```

Comment:
So the logic here is
* `DF_BLOCK_SIZE` takes precedence over `BLOCK_SIZE`, so the latter is ignored,
* `DF_BLOCK_SIZE` is not a valid block size, so it is ignored,
* the default block size 1K is used.
Is that right?

Reply:
Yes, that's correct. I added a new `default_block_size_header` var to make the `assert` a bit less magic.